File: server/src/main/java/com/fluffy/exam/domain/ExamStatus.java
package com.fluffy.exam.domain;

import com.fluffy.global.exception.NotFoundException;
import java.util.Arrays;

public enum ExamStatus {

    DRAFT,
    PUBLISHED,
    ;

    public static ExamStatus from(String value) {
        return Arrays.stream(values())
                .filter(it -> it.name().equalsIgnoreCase(value))
                .findFirst()
                .orElseThrow(() -> new NotFoundException("존재하지 않는 시험 상태입니다. 시험 상태: " + value));
    }

    public boolean isPublished() {
        return this == PUBLISHED;
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/ExamStatus.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.NotFoundException

enum class ExamStatus {

    DRAFT,
    PUBLISHED,
    ;

    companion object {

        fun from(status: String): ExamStatus {
            return entries.find { it.name.equals(status, ignoreCase = true) }
                ?: throw NotFoundException("존재하지 않는 시험 상태입니다. 시험 상태: $status")
        }
    }

    fun isPublished(): Boolean {
        return this == PUBLISHED
    }
}